pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -329,7 +329,8 @@ func (c *Cluster) addOrRemoveExternalMonitor(status client.MonStatusResponse) (b
 		monsNotFound[mon.Name] = struct{}{}
 	}
 
-	monCount := len(status.MonMap.Mons)
+	// monCount := len(status.MonMap.Mons)
+	monCount := len(c.ClusterInfo.Monitors)
```

Comment:
At this point, we cannot use `c.ClusterInfo.Monitors`, because we have no idea how it was populated. So relaying on `status.MonMap.Mons` is much safer.